internal/databroker/server.go

```diff
@@ -222,10 +219,14 @@ func (srv *Server) Sync(req *databroker.SyncRequest, stream databroker.DataBroke
 		return err
 	}
 
-	ch := srv.onchange.Bind()
-	defer srv.onchange.Unbind(ch)
-	for {
-		updated, _ := db.List(context.Background(), recordVersion)
+	ctx := stream.Context()
+	ch := db.Sync(ctx)
+
+	for range ch {
+		updated, err := db.List(ctx, recordVersion)
```

Comment:
Is there a race condition here from the start of the request to the db.Sync() call?  It looks like we could miss an update as we set up the db.Sync().

Reply:
If something happens before client call `Sync`, then client will be lost that notifications. We can't buffer the notification.

Reply:
I think we should always do the first sync, before we going to listen to the channel.

Reply:
Yes I was thinking something like that.  Either have the db call return a signal once at the start to force the caller to catch up, or have this code do a list after the channel is set up to pick up anything in between.